### PR TITLE
Fix checking whether user has access to the group in og_context()

### PR DESCRIPTION
**Pull request by HelenaEksler**
1. Check if we already have context only if we don't have to check user's permissions to access the current group.
2. Check whether user has access to the group if it's necessary.

### CODE REVIEW

File: og_context/og_context.module [hunk 1]
```diff
@@ -200,24 +200,43 @@ function og_context($group_type = 'node', $group = NULL, $account = NULL, $check
...
+    $group_type,
+    $id,
+    $account->uid,
+    $check_access,
+  );
+  $cache_key = implode(':', $cache_keys);
+
+  $context = &drupal_static(__FUNCTION__, array());
+
+  if (empty($context[$cache_key])) {
```
Comment (amitaibu) [inline og_context/og_context.module]: this looks wrong - if we called `og_context()` and it was empty, then it will enter here again in the second call.

**Comment by amitaibu** [comment]
Merged, thanks

File: og_context/og_context.module [hunk 2]
```diff
@@ -199,27 +199,44 @@ function og_context($group_type = 'node', $group = NULL, $account = NULL, $check
...
-        return $context;
-      }
+  $id = FALSE;
+  if ($group) {
+    list($id) = entity_extract_ids($group_type, $group);
+  }
+
+  $cache_keys = array(
+    $group_type,
+    $id,
```
Comment (pianomansam) [inline og_context/og_context.module:210]: This right here is the cause of not being able to retrieve the [current active groups](https://www.drupal.org/node/2720337). It makes $id required from $group, even if we just want a list of the active groups. What happens on og_context set is that $cache_key becomes something like this - `node:123:0:TRUE`

Then when trying to retrieve a list of active groups with `og_context()`, the $cache_key becomes `node::0:TRUE` which **never** matches anything, thus always returning false.

I wonder if we need separate functions for setting, getting, and getting active og contexts. Perhaps something like this:
- `function og_context_set($group_type, $group)`
- `function og_context_get($group_type = 'node', $group = NULL, $account = NULL, $check_access = TRUE)`
- `function og_context_get_all($group_type = 'node')`

**Comment by amitaibu** [comment]
@pianomansam Thanks. worth opening a new issue/ PR

**Pull request closed without merging**